litex/tools/litex_json2dts_linux.py

```diff
@@ -44,18 +44,35 @@ def generate_dts(d, initrd_start=None, initrd_size=None, polling=False):
     if initrd_size is None:
         initrd_size = default_initrd_size
 
+    if initrd == "enabled" or initrd is None:
+        initrd_enabled = True
+    elif initrd == "disabled":
+        initrd_enabled = False
+    else:
+        initrd_enabled = True
+        initrd_size = os.path.getsize(initrd)
+
+    if root_device is None:
+        root_device = "ram0"
+
     dts += """
         chosen {{
-            bootargs = "mem={main_ram_size_mb}M@0x{main_ram_base:x} rootwait console=liteuart earlycon=sbi root=/dev/ram0 init=/sbin/init swiotlb=32";
-            linux,initrd-start = <0x{linux_initrd_start:x}>;
-            linux,initrd-end   = <0x{linux_initrd_end:x}>;
-        }};
-""".format(
+            bootargs = "mem={main_ram_size_mb}M@0x{main_ram_base:x} {console} {rootfs} init=/sbin/init swiotlb=32";""".format(
```

Comment:
There's no need to specify `init=/sbin/init`, the kernel will try it first:

https://elixir.bootlin.com/linux/v5.14/source/init/main.c#L1543
```
        if (!try_to_run_init_process("/sbin/init") ||
            !try_to_run_init_process("/etc/init") ||
            !try_to_run_init_process("/bin/init") ||
            !try_to_run_init_process("/bin/sh"))
                return 0;
```

Reply:
I was thinking the same thing for both `mem=` and `init` I dont think its needed.